utils/kafka-components-utils/src/it/scala/pl/touk/nussknacker/engine/kafka/validator/CachedTopicsExistenceValidatorTest.scala

```diff
@@ -1,115 +1,130 @@
-import pl.touk.nussknacker.engine.kafka.validator.TopicsExistenceValidationConfigForTest.createKafkaContainer
-import pl.touk.nussknacker.engine.kafka.{
-  CachedTopicsExistenceValidatorConfig,
-  KafkaConfig,
-  KafkaUtils,
-  TopicsExistenceValidationConfig
-}
+import pl.touk.nussknacker.engine.kafka._
 
 import java.util.Collections
 import scala.concurrent.duration.DurationInt
 
-object TopicsExistenceValidationConfigForTest {
-
-  val config: TopicsExistenceValidationConfig = {
-    // longer timeout, as container might need some time to make initial assignments etc.
-    TopicsExistenceValidationConfig(
-      enabled = true,
-      validatorConfig = CachedTopicsExistenceValidatorConfig.DefaultConfig.copy(adminClientTimeout = 5 seconds)
-    )
-  }
-
-  def createKafkaContainer(): KafkaContainer =
-    KafkaContainer(DockerImageName.parse(s"${KafkaContainer.defaultImage}:7.4.0"))
-}
-
 class CachedTopicsExistenceValidatorWhenAutoCreateDisabledTest
-    extends AnyFunSuite
-    with ForAllTestContainer
-    with Matchers {
-
-  override val container: KafkaContainer =
-    createKafkaContainer().configure(_.withEnv("KAFKA_AUTO_CREATE_TOPICS_ENABLE", "FALSE"))
-
-  private def kafkaConfig = KafkaConfig(
-    kafkaProperties = Some(Map("bootstrap.servers" -> container.bootstrapServers)),
-    kafkaEspProperties = None,
-    consumerGroupNamingStrategy = None,
-    avroKryoGenericRecordSchemaIdSerialization = None,
-    topicsExistenceValidationConfig = TopicsExistenceValidationConfigForTest.config
-  )
+    extends BaseCachedTopicsExistenceValidatorTest(
+      kafkaAutoCreateEnabled = false
+    ) {
 
   test("should validate existing topic") {
-    val topic = new NewTopic("test.topic.1", Collections.emptyMap())
-    KafkaUtils.usingAdminClient(kafkaConfig) {
-      _.createTopics(Collections.singletonList[NewTopic](topic))
-    }
-    val v = new CachedTopicsExistenceValidator(kafkaConfig)
-    v.validateTopic(TopicName.ForSource(topic.name())) shouldBe Symbol("valid")
+    val topic     = createSourceTopic("test.topic.1")
+    val validator = new CachedTopicsExistenceValidator(defaultKafkaConfig)
+    validator.validateTopic(topic) shouldBe Valid(topic)
   }
 
   test("should validate not existing topic") {
-    val v = new CachedTopicsExistenceValidator(kafkaConfig)
-    v.validateTopic(TopicName.ForSource("not.existing")) shouldBe Symbol("invalid")
+    val validator = new CachedTopicsExistenceValidator(defaultKafkaConfig)
+    validator.validateTopic(notExistingSourceTopic) shouldBe Invalid(
+      TopicExistenceValidationException(NonEmptyList.one(notExistingSourceTopic))
+    )
   }
 
   test("should not validate not existing topic when validation disabled") {
-    val v = new CachedTopicsExistenceValidator(
-      kafkaConfig.copy(
+    val validator = new CachedTopicsExistenceValidator(
+      defaultKafkaConfig.copy(
         kafkaProperties = Some(Map("bootstrap.servers" -> "broken address")),
```

Comment:
i think this broken address is redundant and makes test fail for the wrong reason - `Failed to create new KafkaAdminClient`. but after commenting it out it fails for the right reason.